experiments/scripts/demoChecker.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Default values
+INPUT_DIR_COUNT=0
+INPUT_DIRS=()
+OUTPUT_DIR=$HOME/.demoChecker
+TIMEOUT_DURATION=30
+TIMEOUT_KILL_DURATION=60
+
+# Command line arguments
+for arg in "$@"
+do
+    case $arg in
+        -c|--input-count)
+            INPUT_DIR_COUNT="$2"
+            shift
+            shift
+        ;;
+        -dirs|--directories)
+            shift
+            for counter in $(seq 1 $INPUT_DIR_COUNT)
+            do
+                INPUT_DIRS+=("$1")
+                shift
+            done
+        ;;
+        -o|--output)
+            OUTPUT_DIR="$2"
+            shift
+            shift
+        ;;
+        -t|--timout-duration)
+            TIMEOUT_DURATION="$2"
+            shift
+            shift
+        ;;
+        -tk|--timeout-kill-duration)
+            TIMEOUT_KILL_DURATION="$2"
+            shift
+            shift
+        ;;
+    esac
+done
+
+# Prepare pass and fail files
+if [ ! -d $OUTPUT_DIR ]; then
+    mkdir -p $OUTPUT_DIR
+fi
+if [ -f "$OUTPUT_DIR/passFile" ]; then
+    rm "$OUTPUT_DIR/passFile"
+    touch "$OUTPUT_DIR/passFile"
+fi
+if [ -f "$OUTPUT_DIR/failFile" ]; then
+    rm "$OUTPUT_DIR/failFile"
+    touch "$OUTPUT_DIR/failFile"
+fi
+
+echo "Output Directory: " $OUTPUT_DIR
+
+# Run through the executables in the INPUT_DIRS
+for dir in "${INPUT_DIRS[@]}"
+do
+    echo "Dir: " $dir
+    pushd $dir
+    for file in $(find . -executable -type f)
+    do
+        echo "File: " $file
+        timeout -k $TIMEOUT_KILL_DURATION $TIMEOUT_DURATION $file
```

Comment:
I don't think this will work with the Khronos samples, which require the sample name as an argument?